src/eonstor_ds_cli/common_cli.py

```diff
@@ -496,26 +457,40 @@ def _create_partition_with_pool(
             extraspecs = self._get_extraspecs_dict(volume['volume_type_id'])
 
         provisioning = self._get_extraspecs_value(extraspecs, 'provisioning')
-        tiering = self._get_extraspecs_value(extraspecs, 'tiering')
+        self._check_extraspec_value(provisioning, self.provisioning_values)
+        top_tier = self._get_top_tier(pool_id)
 
         extraspecs_dict = {}
         cmd = ''
-        if provisioning == 'thin':
-            provisioning = int(volume_size * 0.2)
-            extraspecs_dict['provisioning'] = provisioning
-            extraspecs_dict['init'] = 'disable'
+        if top_tier == '-1':
+            if provisioning == 'thin':
+                provisioning = int(volume_size * 0.2)
+                extraspecs_dict['provisioning'] = provisioning
+                extraspecs_dict['init'] = 'disable'
         else:
-            self._check_extraspec_value(
-                provisioning, self.provisioning_values)
-
-        if tiering != '0':
-            self._check_extraspec_value(
-                tiering, self.tiering_values)
-            tier_levels_list = list(range(int(tiering)))
-            tier_levels_list = list(map(str, tier_levels_list))
-            self._check_tiering_existing(tier_levels_list, pool_id)
-            extraspecs_dict['provisioning'] = 0
-            extraspecs_dict['init'] = 'disable'
+            tiering = self._get_extraspecs_value(extraspecs, 'tiering')
+            if provisioning == 'full':
+                if tiering != '-1':
+                    tier_levels_list = tiering.split(',')
+                    tier_levels_list = list(map(str, tier_levels_list))
+                    self._check_tiering_existing(tier_levels_list, pool_id)
+                    if len(tier_levels_list) > 1:
+                        msg = _('Must specify only one tier instead of '
+                                '%(tier_levels_list)s tier(s)') % {
+                            'tier_levels_list': tier_levels_list}
+                        LOG.error(msg)
+                        raise exception.VolumeDriverException(message=msg)
+                    extraspecs_dict['tiering'] = tiering
+                else:
+                    extraspecs_dict['tiering'] = top_tier
+            else:
+                extraspecs_dict['provisioning'] = 0
+                extraspecs_dict['init'] = 'disable'
+                if tiering != '-1':
+                    tier_levels_list = tiering.split(',')
+                    tier_levels_list = list(map(str, tier_levels_list))
+                    self._check_tiering_existing(tier_levels_list, pool_id)
+                    extraspecs_dict['tiering'] = tiering
```

Comment:
Please refactor and create a function for checking pool whether fit extraspec or not.

This place should get extraspec value only.

For example.
Create Volume => Get the fit pool or Check extraspec fit or not => Create patition with pool

The same as migrate volume.